launchdarkly/environment_utils.go

```diff
@@ -0,0 +1,120 @@
+package launchdarkly
+
+import (
+	"strconv"
+)
+
+const dummyEnvironmentKey = "dummy-environment"
+
+func getEnvironmentKeys(client Client, project string) ([]string, error) {
+	payload := make(map[string]interface{})
+
+	err := client.Get("/projects/"+project, map[int]bool{200: true}, &payload)
+	if err != nil {
+		return nil, err
+	}
+
+	environments := payload["environments"].([]interface{})
+
+	keys := make([]string, len(environments))
+
+	for i, env := range environments {
+		keys[i] = env.(map[string]interface{})["key"].(string)
+	}
+
+	return keys, nil
+}
+
+func ensureWeCanDeleteEnvironment(client Client, project string) error {
+	onlyOne, err := isThereOnlyOneEnvironment(client, project)
+	if err != nil {
+		return err
+	}
+
+	if onlyOne {
+		println("Creating dummy environment since we cannot delete the last environment in a project")
+		return ensureThereIsADummyEnvironment(client, project)
+	} else {
+		return nil
+	}
+}
+
+func ensureThereIsADummyEnvironment(client Client, project string) error {
+	exists, err := isThereADummyEnvironment(client, project)
+	if err != nil {
+		return err
+	}
+
+	if !exists {
+		return createDummyEnvironment(client, project)
+	} else {
+		return nil
+	}
+}
+
+func ensureThereIsNoDummyEnvironment(client Client, project string) error {
+	exists, err := isThereADummyEnvironment(client, project)
+	if err != nil {
+		return err
+	}
+
+	if exists {
+		println("A dummy environment was found, deleting it")
+		return deleteDummyEnvironment(client, project)
+	} else {
+		println("No dummy environment was found")
+		return nil
+	}
+}
+
+func isThereADummyEnvironment(client Client, project string) (bool, error) {
+	statusCode, err := client.GetStatus("/projects/" + project + "/environments/" + dummyEnvironmentKey)
+	if err != nil {
+		return false, err
+	}
+
+	return statusCode == 200, nil
```

Comment:
This is the only place `statusCode` is used. Why not directly return the payload and here, you could simply verify that err is nil and that there is a payload.

Reply:
Alright, I guess.

Reply:
Well in fact, I don't think so. It's way more accurate to check for the explicit status, since an error can be raised if the returned status is unexpected, or if the call failed for another reason.